doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -441,80 +425,140 @@ automatically created.
-
-      This sets an optional mode of use of the identifier :n:`@qualid`. When
-      proof-search faces a goal that ends in an application of :n:`@qualid` to
-      arguments :n:`@term ... @term`, the mode tells if the hints associated to
-      :n:`@qualid` can be applied or not. A mode specification is a list of n ``+``,
+
+      .. warning::
+
+         There is no operator precedence during parsing, one can
+         check with :cmd:`Print HintDb` to verify the current cut expression.
+
+      .. warning::
+
+         These hints currently only apply to typeclass proof search and the
+         :tacn:`typeclasses eauto` tactic.
+
+   .. cmd:: Hint Mode @qualid {+ {| + | ! | - } } {? : {+ @ident } }
+
+      Sets an optional mode of use for the identifier :n:`@qualid`. When
+      proof search has a goal that ends in an application of :n:`@qualid` to
+      arguments :n:`@arg ... @arg`, the mode tells if the hints associated with
+      :n:`@qualid` can be applied or not. A mode specification is a list of ``+``,
       ``!`` or ``-`` items that specify if an argument of the identifier is to be
       treated as an input (``+``), if its head only is an input (``!``) or an output
       (``-``) of the identifier. For a mode to match a list of arguments, input
       terms and input heads *must not* contain existential variables or be
-      existential variables respectively, while outputs can be any term. Multiple
-      modes can be declared for a single identifier, in that case only one mode
-      needs to match the arguments for the hints to be applied. The head of a term
+      existential variables respectively, while outputs can be any term.
+
+      The head of a term
       is understood here as the applicative head, or the match or projection
       scrutinee’s head, recursively, casts being ignored. :cmd:`Hint Mode` is
       especially useful for typeclasses, when one does not want to support default
       instances and avoid ambiguity in general. Setting a parameter of a class as an
-      input forces proof-search to be driven by that index of the class, with ``!``
-      giving more flexibility by allowing existentials to still appear deeper in the
-      index but not at its head.
+      input forces proof search to be driven by that index of the class, with ``!``
+      allowing existentials to appear in the index but not at its head.
 
    .. note::
 
-      + One can use a :cmd:`Hint Extern` with no pattern to do
-        pattern matching on hypotheses using ``match goal with``
-        inside the tactic.
+      + Multiple modes can be declared for a single identifier.  In that
+        case only one mode needs to match the arguments for the hints to be applied.
 
       + If you want to add hints such as :cmd:`Hint Transparent`,
         :cmd:`Hint Cut`, or :cmd:`Hint Mode`, for typeclass
         resolution, do not forget to put them in the
         ``typeclass_instances`` hint database.
 
+.. _hintrewrite:
```

Comment:
This label is only used once (earlier in this file). I would remove it and refer to the command instead.